src/back/spv/writer.rs

```diff
@@ -1772,8 +1786,28 @@ impl Writer {
             .to_words(&mut self.logical_layout.ext_inst_imports);
 
         if self.flags.contains(WriterFlags::DEBUG) {
-            self.debugs
-                .push(Instruction::source(spirv::SourceLanguage::GLSL, 450));
+            if let Some(file_name) = (|| {
+                Path::new(ir_module.file_path.as_ref()?)
+                    .file_name()?
+                    .to_str()
+            })() {
```

Comment:
Please avoid those immediately called closures - I don't see why they are needed.

Reply:
This closure is a workround for the try block. I know this is a bit ugly, but I want to avoid nested `if let Some` checks for Option. I don't have any better idea yet.

Reply:
I think we can use `and_then` and `map`.